components/PullRequests/PrStateAuthorIcon.tsx

```diff
@@ -0,0 +1,63 @@
+import { GitMergeIcon, GitPullRequestClosedIcon, GitPullRequestIcon } from "@primer/octicons-react";
+import { Avatar } from "components/atoms/Avatar/avatar-hover-card";
+
+function getPullRequestStateIcon(
+  state: DbRepoPREvents["pr_state"],
+  isDraft: DbRepoPREvents["pr_is_draft"],
+  isMerged: DbRepoPREvents["pr_is_merged"]
+) {
+  const size = 16;
+
+  switch (true) {
+    case state === "open" && !isDraft:
+      return <GitPullRequestIcon aria-label="open pull request" size={size} className="text-white" />;
+    case state === "open" && isDraft:
+      return <GitPullRequestIcon size={size} aria-label="draft pull request" className="text-white" />;
+    case state === "closed" && !isMerged:
+      return <GitPullRequestClosedIcon size={size} aria-label="closed pull request" className="text-white" />;
+    case state === "closed" && isMerged:
+      return <GitMergeIcon size={size} aria-label="merged pull request" className="text-white" />;
+  }
+}
+
+export const PrStateAuthorIcon = ({
+  state,
+  isDraft,
+  isMerged,
+  author,
+}: {
+  state: string;
+  isDraft: boolean;
+  isMerged: boolean;
+  author: string;
+}) => {
+  let backgroundColor = "";
+
+  switch (true) {
+    case state === "open" && !isDraft:
+      backgroundColor = "bg-green-600";
+      break;
+
+    case state === "open" && isDraft:
+      backgroundColor = "bg-slate-600";
+      break;
+    case state === "closed" && !isMerged:
+      backgroundColor = "bg-red-600";
+      break;
+    case state === "closed" && isMerged:
+      backgroundColor = "bg-purple-600";
+      break;
+  }
+
+  return (
+    <div className="relative w-max">
+      <Avatar contributor={author} size="medium" />
+      <div
+        className={`absolute -bottom-2 -right-2 p-1 rounded-full [&_svg]:absolute [&_svg]:top-[3.5px] [&_svg]:left-[3.5px] ${backgroundColor}`}
```

Comment:
absolute nitpick on the style side, but I'd love to make some tiny changes to get here: 
![image](https://github.com/open-sauced/app/assets/27322879/e8d68360-b8a1-4e0f-8e9c-23aa4cae2759)

move the icon circle a bit more to the bottom and to the right (had to hardcode values bc tailwind's values didn't work) and add a 2px white border to make it feel like it's separated from the avatar

` className={absolute -bottom-[-10px] -right-[-12px] p-1 border-[2px] border-white  rounded-full [&_svg]:absolute [&_svg]:top-[3.5px] [&_svg]:left-[3.5px] `